### PR TITLE
Servlet delete comments

### PR DESCRIPTION
**Pull request by RolandNaijuka**
-Add input so that user can select max number of comments to display
-Add button to delete all comments

[Link to demo](https://rnaijuka-step-2020.ue.r.appspot.com)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,12 +32,37 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private final int MAX_COMMENTS = 3;
-    private ArrayList<Comments> userComments = new ArrayList<>();
     
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
```
Comment (vitashubin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:37]: nit: you don't have to have an empty space at the beginning of the function

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -27,12 +32,37 @@
...
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        int numComments;
+        try{
+            numComments = Integer.parseInt(request.getParameter("numComments"));
+        }catch(NumberFormatException e){
+            // Default number of comments
+            numComments = 5;
+        }
```
Comment (vitashubin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: nit: style guide usually puts spaces before the { }s and words 
so:
try {
 .....
} catch(....) {
}

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        Query query = new Query("Comments");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        for(Entity entity: results.asIterable()){
+            datastore.delete(entity.getKey());
+        }
+        response.sendRedirect("/contact.html");
+    }
+}
```
Comment (vitashubin) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java]: newline :)

File: portfolio/src/main/webapp/contact.html [hunk 4]
```diff
@@ -27,16 +27,24 @@ <h1>Contact Me!</h1>
...
+            <button role="submit" name="commentButton">Submit</button>
 		</form>
+        <form>
+            <div>
+                <label>Number of Comments: </label>
+                <input type="number" min="0" max="50" id="numComments" onchange="updateNumComments();">
+            </div>
+        </form>
+        <button onclick="deleteComments();">Delete Comments<button>
 		<fieldset id="user-comments">
```
Comment (vitashubin) [inline portfolio/src/main/webapp/contact.html]: If this fieldset is empty, is it still needed?

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -43,19 +55,45 @@ async function getUserComments(){
...
 }
 
+/* Delete all the comments from the server */
+async function deleteComments(){
+    try{
+        await fetch("/delete-data", {
+            method: "POST"
+        });
+    }
+    catch(error){
```
Comment (vitashubin) [inline portfolio/src/main/webapp/script.js]: nit: have the catch on the same line as the closing bracket and add spaces 
try {
 ...
} catch(...) {
} finally {
}

**Review by vitashubin — APPROVED** [review]

File: portfolio/src/main/webapp/contact.html (hunk 4 above)
Comment (RolandNaijuka) [reply portfolio/src/main/webapp/contact.html]: > If this fieldset is empty, is it still needed?

I need to store the comments. But I guess I can create it in js.

**Pull request merged**